rpcserver.go

```diff
@@ -6745,6 +6751,222 @@ func (r *rpcServer) ListPermissions(_ context.Context,
+		}
+	}
+
+	// Look for any inconsistency trying to add AND remove the same address
+	for _, addr := range removeAddr {
+		if findAddr(addr, addAddr) {
+			return nil, nil, fmt.Errorf("invalid updates for removing AND adding %v", addr)
+		}
+	}
+
+	ops := &lnrpc.Op{Entity: "addresses"}
+	newAddrs := make([]net.Addr, 0, len(updates)+len(currentAddresses))
+
+	// Copy current addresses excluding the ones that need to be removed
+	for _, addr := range currentAddresses {
+		if findAddr(addr, removeAddr) {
+			ops.Actions = append(ops.Actions, fmt.Sprintf("%s removed", addr.String()))
+			continue
+		}
+		newAddrs = append(newAddrs, addr)
+	}
+
+	// Add new adresses if needed
+	for _, addr := range addAddr {
+		if !findAddr(addr, newAddrs) {
+			ops.Actions = append(ops.Actions, fmt.Sprintf("%s added", addr.String()))
+			newAddrs = append(newAddrs, addr)
+		}
+	}
+
+	return newAddrs, ops, nil
+}
+
+// 	UpdateNodeAnnouncement allows the caller to update the node parameters
+//  and broadcasts a new version of the node announcement to its peers.
+func (r *rpcServer) UpdateNodeAnnouncement(_ context.Context,
+	req *lnrpc.NodeAnnouncementUpdateRequest) (
+	*lnrpc.NodeAnnouncementUpdateResponse, error) {
+
+	resp := &lnrpc.NodeAnnouncementUpdateResponse{}
+	nodeModifiers := make([]netann.NodeAnnModifier, 0)
+	currentNodeAnn, err := r.server.genNodeAnnouncement(false)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get current node announcement: %v", err)
+	}
+
+	if len(req.FeatureUpdates) > 0 {
+		ops, err := r.updateFeatures(req.FeatureUpdates)
+		if err != nil {
+			return nil, fmt.Errorf("error trying to update node features: %w", err)
+		}
+		resp.Ops = append(resp.Ops, ops)
+	}
+
+	if req.Color != "" {
+		color, err := parseHexColor(req.Color)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse color: %v", err)
+		}
+
+		if color != currentNodeAnn.RGBColor {
+			resp.Ops = append(resp.Ops, &lnrpc.Op{
+				Entity:  "color",
+				Actions: []string{fmt.Sprintf("changed to %v", color)}})
+			nodeModifiers = append(nodeModifiers, netann.NodeAnnSetColor(color))
+		}
+	}
+
+	if req.Alias != "" {
+		alias, err := lnwire.NewNodeAlias(req.Alias)
+		if err != nil {
+			return nil, fmt.Errorf("invalid alias value: %v", err)
+		}
+		if alias != currentNodeAnn.Alias {
+			resp.Ops = append(resp.Ops, &lnrpc.Op{
+				Entity:  "alias",
+				Actions: []string{fmt.Sprintf("changed to %v", alias)}})
+			nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAlias(alias))
+		}
+	}
+
+	if len(req.AddressUpdates) > 0 {
+		newAddrs, ops, err := r.updateAddresses(currentNodeAnn.Addresses, req.AddressUpdates)
+		if err != nil {
+			return nil, fmt.Errorf("error trying to update node addresses: %w", err)
+		}
+		resp.Ops = append(resp.Ops, ops)
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAddrs(newAddrs))
+	}
+
+	if len(nodeModifiers) == 0 {
+		return nil, fmt.Errorf("unable detect any new values to update " +
```

Comment:
When only updating features, UpdateNodeAnnouncement returns an error (but still updates the featurebit) .

Reply:
Thank you for noticing @sputn1ck 

Actually this can be tricky because even if when we have updated some parts of the node it is not until the end of the function that we update the on-disk version of our announcement so other sub-systems don't become out of sync + announce those changes to our peers.

I will return `resp` instead of `nil` so the user always have track of what operations were performed. Even so, if some operations succeed but the endpoint fails before completing the full flow the user will have to call it again until it succeeds to ensure that the copy on disk + our peers have the correct node announcement version